.gitignore

```diff
@@ -1,6 +1,3 @@
-# mlx-graphs specific ignores
-data/
```

Comment:
I put this one as GCN and GAT examples download the Cora dataset into a data/ folder, and it's cleaner to add this one to .gitignore.

Reply:
> I put this one as GCN and GAT examples download the Cora dataset into a data/ folder, and it's cleaner to add this one to .gitignore.

Yeah, the problem is that we have modules and files with the `data` pattern that would be gitignored with that line. I suggest renaming the dataset folder to something different (possibly a dotfile).